src/components/HomePage/homePage.js

```diff
@@ -0,0 +1,91 @@
+import "bootstrap/dist/css/bootstrap.min.css";
+import { Routes, Route } from "react-router-dom";
+import AboutUs from "../AboutUs/aboutUs";
+import ContactUs from "../ContactUs/contactUs";
+import Message from "../Message/message";
+import Messages from "../Messages/messages";
+import messages from "../../message.json";
+import { useState, useEffect } from "react";
+import "./homePage.css";
+import { Link } from "react-router-dom";
+
+const HomePage = () => {
+  const [activeAbout, setActiveAbout] = useState(false);
+  const [activeContact, setActiveContact] = useState(false);
+  const [activeMessages, setActiveMessages] = useState(false);
+
+  const aboutColorChange = () => {
+    setActiveAbout(true);
+    setActiveMessages(false);
+    setActiveContact(false);
+  };
+  const contactColorChange = () => {
+    setActiveContact(true);
+    setActiveMessages(false);
+    setActiveAbout(false);
+  };
+  const messagesColorChange = () => {
+    setActiveMessages(true);
+    setActiveContact(false);
+    setActiveAbout(false);
+  };
+
+  useEffect(() => {}, []);
+
+  return (
+    <>
+      <div className="navbar-link-list">
+        <div>
+          <h3>
+            {" "}
+            <Link
+              to="/about"
+              id="about"
+              className={activeAbout ? "color" : "real-color"}
```

Comment:
you do not need all this stuff to do. Read https://reacttraining.com/react-router/web/api/NavLink the change the color of Active Navigation. Or YOu can use useHistory hook to add conditions like this. @PuniaMohit